### DISCUSSION TITLE
How to change content of a label

### DISCUSSION THREAD

**Discussion by ceuser1** [serilog-contrib/serilog-sinks-grafana-loki#200] [General]
At the app startup I configure the logger, so I can log exceptions as soon as possible. One of the labels is `User`, which is unknown at this time. 

For example:

```csharp
	SelfLog.Enable(p => Debug.WriteLine(p));
	Log.Logger = new LoggerConfiguration()
		.MinimumLevel.Verbose()
		.Enrich.With(new DayEnricher())
		.WriteTo.GrafanaLoki(
			Settings.Default.LogLokiEndpoint,
			new List<LokiLabel>
			{
				new() { Key = "Country", Value = "MyCountry" },
				new() { Key = "Company", Value = "MyCompany" },
				new() { Key = "Project", Value = "MyProject" },
				new() { Key = "Application", Value = "MyApplication" },
				new() { Key = "User", Value = "unknown at this time" },
				new() { Key = "Terminal", Value = Environment.MachineName },
				new() { Key = "Ip", Value = LoggingHelper.GetLocalIpAddress() },
				new() { Key = "Version", Value = LoggingHelper.GetApplicationVersion() },
				new() { Key = "Culture", Value = "en-US" },
				new() { Key = "ApplicationLocation", Value = LoggingHelper.GetApplicationLocation() }
			},
			new List<string> { "Day", "Tag" },
			credentials: new LokiCredentials { Login = Settings.Default.LogLokiUsername, Password = Settings.Default.LogLokiPassword },
			restrictedToMinimumLevel: Settings.Default.LogLokiSinkMinLevel
		)
		.WriteTo.Logger(p => p
			.WriteTo.File(
				Settings.Default.LogFileSinkFolder,
				Settings.Default.LogFileSinkMinLevel,
				"{Timestamp:yyyy-MM-dd HH:mm:ss.fff zzz} [{Level:u3}]|{Message:lj}{NewLine}{Exception}",
				rollingInterval: RollingInterval.Day
			)
		)
		.CreateLogger();
```

Then the user logs in, so I know his `User`. Is it possible to overwrite the `User` label, or do I have to configure the full logger again?

**Comment by mishamyte** [comment]
Hi! 

I would not recommend to overuse the static labels (properties), cause it will lead to the [high cardinality problem](https://github.com/serilog-contrib/serilog-sinks-grafana-loki/wiki/Common-problems#high-cardinality-problems).

If you use logger as it was designed, for [the structured logging](https://github.com/serilog/serilog/wiki/Structured-Data), you will have all your labels in place.

Also, probably, you should read [that page from the documentation](https://github.com/serilog-contrib/serilog-sinks-grafana-loki/wiki/Using-LokiJsonTextFormatter-and-Grafana-Loki-json-parser). It has an examples how to work with such kind of data, using Grafana's JSON parser as a part of the pipeline step

**Comment by ceuser1** [comment]
Thanks @mishamyte! I removed most of the labels and started using structured logging with the combination of JSON parser, and it works perfect :)

And for my original question - instead of a label for `User` I use a property: `LogContext.PushProperty("User", user.Name);`
